### ISSUE TITLE
Custom unit/building bugs

### ISSUE THREAD

**Issue by GokuSS400** [declantyson/civ6-Zelda_Hyrule#9]
I just downloaded this mod and played a game with it, I've gotten to the renaissance era thus far but i've noticed a number of issues that need to be addressed....

- Upgrading Warriors to Hero of Time costs about 1.5k Gold.  Two possibilities emerge to fix this....either reduce the cost by a factor of 10 or something else like expending faith to upgrade them

- Hero of Time requires 2,000 FAITH to recruit from a City.  Since this unit replaces the SWORDSMAN which is an early game unit, making this essentially pointless unless you are not doing anything with your Faith throughout the game.

- Hero of Time custom unit cannot be upgraded even after researching the next level (i.e. cannot upgrade to Musketmen and so on)

Edit: removed a topic and reposting it as a separate thread

**Comment by lapost48** [comment]
Please put each bullet in a separate issue so they can be addressed and closed individually.

- Upgrade cost is a side effect, the intention of the Hero of Time is that you can't purchase or build them. You should only get one by advancing to the next era (Only ever one Link in any game, they are also immensely more powerful than Swordsmen and I believe Musketmen). This is also why the Hero cannot be upgraded to a Musketman. I believe the  Hero of Time issues are design choices rather than bugs.

-I will look into the Temp[le of Time bug once I have more free time.

**Comment by GokuSS400** [comment]
I can edit this and repost each thing if you really need me to....though 3/4 things relate to the Hero of Time Unit.

Also, are they able to be upgraded to infantry? (I checked the wiki and the Hero of Time unit would be on par with the Musketman unit since it has 55 attack, so it would probably be best to double check they can be upgraded to infantry)

**Comment by lapost48** [comment]
Ideally each issue relates to a single fix, so Temple of Time issue should defnintely be separate.

Each issue should be separate if they are different issues. For example, Upgrade cost from warrior to HoT is high would be one issue; however, the faith cost is directly linked to the cost of the unit so those issues should be together under a single issue called "Hero of Time cost" or whatever you would title the issue.

Not being able to upgrade the unit would be a separate issue called "Unable to Upgrade Hero of Time". These are issues that can be fixed independantly of each other and separating the issues allows for more clarity when the bugs are fixed, rather than having one issue that is half fixed.

To address the issues while I am making this long post, I believe that, ideally, the Hero of Time should not replace the Swordsman, but rather just be a uniqe unit that is granted to the player on new eras. No cost associated and no upgrade progression. Then you keep the Swordsman and that progresses as normal. That solution would stay true to the original intention as well as solve the issues you presented.

Does this seem like a reasonable fix?

I don't have much free time at the moment, but when I do I will get around to working on these issues.

**Comment by GokuSS400** [comment]
Yes, that does sound like a "best of both" balance idea....assuming it can be done (I honestly don't know anything about Civ 6 modding at all).

Thanks for your help!!! (I will make sure to post the Temple of Time issue as a separate thread);

Also no worries, I'm not in any rush to see this done like RIGHT THIS SECOND haha....I do think this is definitely a really cool mod.

Yeah definitely tinker around with the fixes and stuff when you have the time.
